lib/internal/streams/operators.js

```diff
@@ -159,7 +159,7 @@ function map(fn, options) {
   }.call(this);
 }
 
-async function* asIndexedPairs(options) {
+async function* asIndexedPairs(options = {}) {
```

Comment:
We don't need to create a throw away object here.
```suggestion
async function* asIndexedPairs(options = undefined) {
```